src/Driver.cpp

```diff
@@ -311,10 +317,67 @@ int Driver::checkNotificationCommandMode(string const& buffer) const
 {
     if (buffer.size() < 4)
         return 0;
+    // All 4 letters notification
     else
-    {   // All 4 letters notification
-        if (buffer.find("RECV") !=string::npos ||
-                buffer.find("DELI") !=string::npos ||
+    {   // If Notification is a Received Message, the message part of string may contain malicious characters.
+        // RECVxxx,<length>,<source address>,<destination address>,...<data><end-line>
+        // <length> is size of <data>
+        if (buffer.substr(0,4).find("RECV") !=string::npos)
+        {
+            if(buffer.size() < 7)
+                return 0;
+            Notification notification = usblParser.findNotification(buffer);
+            if( notification == RECVIM ||
+                notification == RECVIMS ||
+                notification == RECVPBM )
+            {
+                // Get the amount of comma expected for a notification
```

Comment:
Best would be to put this code in a separate checkIMNotification() (equivalent of checkRegularResponse)